src/Subscriptions/Repositories/SubscriptionRepository.php

```diff
@@ -163,6 +177,12 @@ public function insert(Subscription $subscription)
             $subscription->bumpRenewalDate();
         }
 
+        // If the subscription is not associated with a campaign, try to find the campaign ID by the form ID
+        if (!$subscription->campaignId) {
+            $campaign = give()->campaigns->getByFormId($subscription->donationFormId);
+            $subscription->campaignId = $campaign ? $campaign->id : null;
```

Comment:
I don’t think we are prepared to handle a nullable `campaignId`, which means that if we can’t find a campaign through the `donationFormId`, we might face a fatal error here. Maybe default to `0`?

Reply:
Hmm, we are doing the same thing for donations.